app/src/main/java/com/github/polybooks/activities/SaleInformationActivity.kt

```diff
@@ -20,31 +27,61 @@ class SaleInformationActivity: AppCompatActivity() {
+        findViewById<TextView>(R.id.sale_information_authors).text = StringsManip.listAuthorsToString(sale.book.authors)
         // val countryFlag: TextView = findViewById(R.id.countryFlag)
-        val viewPublishDate: TextView = findViewById(R.id.sale_information_book_publish_date)
-        val viewPublisher: TextView = findViewById(R.id.sale_information_book_publisher)
-        val viewFormat: TextView = findViewById(R.id.sale_information_book_format)
+        findViewById<TextView>(R.id.sale_information_book_format).text = sale.book.format
         // val bookImage = findViewById(R.id.proof_picture)
 
-        val viewCondition: TextView = findViewById(R.id.sale_information_condition)
-        val viewPrice: TextView = findViewById(R.id.sale_information_price)
+        val ratingBar: RatingBar = findViewById(R.id.sale_information_rating)
+        ratingBar.rating = 0f
 
+        if (sale.book.totalStars != null && sale.book.numberVotes != null && sale.book.numberVotes != 0) {
+            ratingBar.rating = (sale.book.totalStars / sale.book.numberVotes).toFloat()
+        }
+        ratingBar.setOnRatingBarChangeListener { bar, rating, _ ->
+            val firestore = FirebaseFirestore.getInstance()
+            val olBookDB = OLBookDatabase { string -> url2json(string) }
+            val bookDB = FBBookDatabase(firestore, olBookDB)
 
-        val sale = (intent.getSerializableExtra(EXTRA_SALE_INFORMATION) as Sale)
+            var query: BookQuery = bookDB.queryBooks()
+            query = query.searchByISBN(setOf(sale.book.isbn))
+            query.getAll().thenCompose { books ->
+                if (books == null || books.isEmpty()) {
+                    throw DatabaseException(
+                        "SaleInformationActivity tried to request information from a Book() " +
+                        "which '${sale.book.isbn}' isbn doesn't exist in database anymore"
+                    )
+                } else if (books.size > 1) {
+                    throw DatabaseException(
+                        "SaleInformationActivity tried to request information from a Book() " +
+                        "which isbn '${sale.book.isbn}' is used ${books.size} times in the database"
+                    )
+                } else {
+                    val book: Book = books[0]
+                    val updatedTotalStars: Double = rating + (book.totalStars ?: 0.0)
+                    val updatedNumberVotes: Int = 1 + (book.numberVotes ?: 0)
 
-        viewTitle.text = sale.book.title
-        viewEdition.text = sale.book.edition
-        viewAuthors.text = StringsManip.listAuthorsToString(sale.book.authors)
+                    // bar.rating = (updatedTotalStars / updatedNumberVotes).toFloat()
+                    bar.setIsIndicator(true)
 
-        viewPublishDate.text = sale.book.publishDate.toString()
+                    bookDB.addBook(
+                        Book(
```

Comment:
you can use the `copy` method on data classes to make a copy and specifying only the fields that change. Like book.copy(isbn = newISBN)